src/devices/src/legacy/rtc_pl031.rs

```diff
@@ -142,10 +142,12 @@ impl RTC {
                 self.trigger_interrupt()?;
             }
             RTCICR => {
-                // As per above mentioned doc, the interrupt is cleared by writing any data value to
-                // the Interrupt Clear Register.
-                self.ris = 0;
-                self.trigger_interrupt()?;
+                // The RTCICR is used together with the RTCMR register to implement a basic time alarm function.
+                // As per the above comment, we do not implement this functionality in firecracker. It can
+                // be easily checked that the value of self.ris will never change. Just like we do
+                // for the RTCMR, we set the expected value and we increment a missed metric.
+                self.ris &= !val;
```

Comment:
what if the interrupt was already 0 and the driver wrote a 0, which should not do anything? this would actually set the interrupt status instead of clearing it

I think we should first do a check that the `val` is not 0, in which case just ignore it, as per the manual

Reply:
So:
self.ris = 0, RTCICR =0, => new self.ris = 0& 1 => 0, so self.ris keeps its value
self.ris = 0, RTCICR=1 => new self.ris = 0 & 0 => so self.ris remains deasserted. 
self.ris =1, RTCICR = 0, new self.ris = 1 & 1 => self.ris keeps its value
self.ris =1, RTCICR = 1, new self.ris = 1 & 0 => self.ris gets deasserted
The bitwise operation has exactly this scope, to keep the value of self.ris if RTCICR is 0 and change it to 0 if RTCICR is 1.

Reply:
I was seeing AND but was reading OR, sorry 😄